sympy/core/expr.py

```diff
@@ -813,11 +813,11 @@ def _eval_interval(self, x, a, b):
             return B - A
 
         value = B - A
-        if all([a.is_comparable, b.is_comparable]):
+        if a.is_comparable and b.is_comparable:
             singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
             for s in singularities:
-                if a < s < b:
-                    value += -limit(self, x, s, "+") + limit(self, x, s, "-")
+                    if a < s < b or b < s < a:
```

Comment:
The two cases `a < b` and `a > b` should be treated differently. (Limits at s from different sides.)